var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -35,113 +32,112 @@ class PyTorchgeo(PythonPackage):
+    depends_on("py-matplotlib@3.3.3:", when="@0.5:", type=("build", "run"))
+    depends_on("py-matplotlib@3.3:", type=("build", "run"))
+    depends_on("py-numpy@1.19.3:", when="@0.5:", type=("build", "run"))
+    depends_on("py-numpy@1.17.2:", type=("build", "run"))
+    depends_on("py-pandas@1.1.3:", when="@0.5:", type=("build", "run"))
+    depends_on("pil@8:", when="@0.5:", type=("build", "run"))
+    depends_on("pil@6.2:", type=("build", "run"))
+    depends_on("pil+jpeg+tiff+zlib", type=("build", "run"))
+    depends_on("py-pyproj@3:", when="@0.5:", type=("build", "run"))
+    depends_on("py-pyproj@2.2:", type=("build", "run"))
+    depends_on("py-rasterio@1.2:", when="@0.5:", type=("build", "run"))
+    depends_on("py-rasterio@1.0.20:", when="@0.3:", type=("build", "run"))
+    depends_on("py-rasterio@1.0.16:", type=("build", "run"))
+    depends_on("py-rtree@1:", when="@0.3:", type=("build", "run"))
+    depends_on("py-rtree@0.9.4:", when="@0.2.1:", type=("build", "run"))
+    depends_on("py-rtree@0.5:", type=("build", "run"))
+    depends_on("py-segmentation-models-pytorch@0.2:", type=("build", "run"))
+    depends_on("py-shapely@1.7.1:", when="@0.5:", type=("build", "run"))
+    depends_on("py-shapely@1.3:", type=("build", "run"))
+    depends_on("py-timm@0.4.12:", type=("build", "run"))
+    depends_on("py-torch@1.12:", when="@0.4:", type=("build", "run"))
+    depends_on("py-torch@1.9:", when="@0.2:", type=("build", "run"))
+    depends_on("py-torch@1.7:", type=("build", "run"))
+    depends_on("py-torchmetrics@0.10:", when="@0.4:", type=("build", "run"))
+    depends_on("py-torchmetrics@0.7:", type=("build", "run"))
+    depends_on("py-torchvision@0.13:", when="@0.4:", type=("build", "run"))
+    depends_on("py-torchvision@0.10:", when="@0.2:", type=("build", "run"))
+    depends_on("py-torchvision@0.3:", type=("build", "run"))
 
     # Optional dependencies
     with when("+datasets"):
-        depends_on("py-h5py@2.6:3", type="run")
-        depends_on("py-laspy@2", when="@0.2:", type="run")
         depends_on("gdal+openjpeg", when="@0.3.1:", type="run")
-        depends_on("libtiff+jpeg+zlib", type="run")
-        depends_on("open3d@0.11.2:0.14+python", when="@0.2:0.3", type="run")
-        depends_on("opencv@3.4.2.17:4+python3+imgcodecs+tiff+jpeg+png", type="run")
-        depends_on("py-pandas@0.23.2:2", when="@0.4.1:", type="run")
-        depends_on("py-pandas@0.23.2:1", when="@0.3:0.4.0", type="run")
-        depends_on("py-pandas@0.19.1:1", when="@0.2", type="run")
-        depends_on("py-pycocotools@2", type="run")
-        depends_on("py-pyvista@0.20:0.38", when="@0.4.1:", type="run")
-        depends_on("py-pyvista@0.20:0.37", when="@0.4.0", type="run")
-        depends_on("py-radiant-mlhub@0.3:0.5", when="@0.4.1:", type="run")
+        depends_on("libtiff+jpeg+libdeflate+zlib+zstd", type="run")
+        depends_on("py-h5py@3:", when="@0.5:", type="run")
```

Comment:
Just for my understanding: 
Why are all dependencies of the variants `run` dependencies and not `("build", "run")`?

Reply:
Technically, the only things that should be build/run are things that appear in both `setup_requires` and `install_requires`. This is usually things like numpy that are either imported in `setup.py` or that we need to link to the libs/headers of.

The only reason that we frequently use build/run for Python/Perl/R in Spack is because it's needed for our import tests. @haampie and I were talking about this and it's likely that we may convert most build/run to run-only in the future. We'll either need to find a way to include run deps when testing, or give up on install-time testing.

For TorchGeo, all optional dependencies are guarded by lazy imports, so they aren't even needed for install-time testing. I would add a comment to this effect, but if we're going to convert to run-only in the future then the comment won't matter.

Reply:
Thanks for the explanation!